### PR TITLE
feat: Button Component

### PR DESCRIPTION
**Pull request by ozairkhan2810**
Four button categories in the component

Please review.
Thanks

link: https://wizardly-dubinsky-67d416.netlify.app/

### CODE REVIEW

File: documentation/index.html [hunk 1]
```diff
@@ -46,10 +42,10 @@ <h3>Avatar</h3>
...
             <p class="text-content">Avatar with different sizes.</p>
             <div class="images">
-              <img class="av-sm mr-2" src="../assets/astro.png" alt="astro" />
-              <img class="av-md mr-2" src="../assets/astro.png" alt="astro" />
-              <img class="av-lg mr-2" src="../assets/astro.png" alt="astro" />
-              <img class="av-xl mr-2" src="../assets/astro.png" alt="astro" />
+              <img class="av av-sm mr-2" src="../assets/astro.png" alt="astro" />
+              <img class="av av-md mr-2" src="../assets/astro.png" alt="astro" />
+              <img class="av av-lg mr-2" src="../assets/astro.png" alt="astro" />
+              <img class="av av-xl mr-2" src="../assets/astro.png" alt="astro" />
```
Comment (AnkitKarnAK) [inline documentation/index.html]: use same name convention either avatar or av

File: documentation/index.html [hunk 2]
```diff
@@ -115,6 +111,36 @@ <h3>Badge</h3>
...
+        <!-- Button Component -->
+        <div id="button" class="ml-2 newline pb-2 mb-2">
+          <h3>Button</h3>
+          <p class="text-content">Primary Button/ Secondary Button</p>
+          <div class="">
+            <button class="btn btn-primary">Primary Button</button>
+            <button class="btn btn-secondary">Secondary Button</button>
+          </div>
+          <p class="text-content">Link Button</p>
+          <div class="">
```
Comment (AnkitKarnAK) [inline documentation/index.html]: no empty classes

File: documentation/index.html [hunk 3]
```diff
@@ -115,6 +111,36 @@ <h3>Badge</h3>
...
+            <button class="btn btn-primary">
+              <a href="">Primary Link Button</a>
+            </button>
+            <button class="btn btn-secondary">
+              <a href="">Secondary Link Button</a>
+            </button>
+          </div>
+          <p class="text-content">Icon Button</p>
+          <div class="">
```
Comment (AnkitKarnAK) [inline documentation/index.html]: no empty classes

File: documentation/index.html [hunk 4]
```diff
@@ -115,6 +111,36 @@ <h3>Badge</h3>
...
+            <button class="btn btn-primary"><img class="icon-img" src="../assets/bell-icon.svg" alt="bell-icon"><span class="btn-text">Hit the Button</span></button>
+            <button class="btn btn-secondary"><img class="icon-img" src="../assets/user-icon.svg" alt="user-icon"><span class="btn-text">Sign In</span></button>
```
Comment (AnkitKarnAK) [inline documentation/index.html]: should be better if img-icon

File: lib/base/variables.css [hunk 5]
```diff
@@ -2,16 +2,8 @@
   --bg_h: #1d2021;
   --bg: #282828;
   --bg_s: #32302f;
```
Comment (AnkitKarnAK) [inline lib/base/variables.css]: should use bg-dark pr bg-light

File: lib/base/variables.css [hunk 6]
```diff
@@ -38,15 +30,8 @@
   --bg: #fbf1c7;
   --bg_s: #f2e5bc;
```
Comment (AnkitKarnAK) [inline lib/base/variables.css]: use same variable name for darker or lighter shades

File: lib/components/avatar.css [hunk 7]
```diff
@@ -1,13 +1,6 @@
 .av {
   height: 4.3rem;
   width: 4.3rem;
```
Comment (AnkitKarnAK) [inline lib/components/avatar.css]: use general size for height & width, 4.3 is bit odd

File: lib/components/avatar.css [hunk 8]
```diff
@@ -1,13 +1,6 @@
...
-}
-
-.av,
-.av-sm,
-.av-md,
-.av-lg,
-.av-xl {
   border-radius: 100%;
```
Comment (AnkitKarnAK) [inline lib/components/avatar.css]: prefer 50%

File: documentation/index.html [hunk 9]
```diff
@@ -115,6 +111,36 @@ <h3>Badge</h3>
...
+            <button class="btn btn-secondary">Secondary Button</button>
+          </div>
+          <p class="text-content">Link Button</p>
+          <div class="">
+            <button class="btn btn-primary">
+              <a href="">Primary Link Button</a>
+            </button>
+            <button class="btn btn-secondary">
+              <a href="">Secondary Link Button</a>
+            </button>
```
Comment (AnkitKarnAK) [inline documentation/index.html]: should be better if you add separate classes for link button

File: lib/components/button.css [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+
+.btn-secondary a,
+.btn-secondary span {
+  color: var(--blue-dim);
+}
+
+.btn-float {
+  height: 2.5rem;
+  width: 2.5rem;
+  border-radius: 100%;
```
Comment (AnkitKarnAK) [inline lib/components/button.css]: prefer 50% for perfect circle

**Review by AnkitKarnAK — COMMENTED** [review]
Commented for some suggestions & for some changes.

**Review by Raghav888 — APPROVED** [review]
Approved

**Review by fahd23 — APPROVED** [review]
Good to merge

**Pull request merged**